Api/Resource.php

```diff
@@ -50,6 +50,34 @@ class Resource implements ResourceInterface
      */
     private $validationGroups;
     /**
+     * @var bool
+     */
+    private $paginationEnabledByDefault = false;
+    /**
+     * @var bool
+     */
+    private $clientAllowedToEnablePagination = false;
+    /**
+     * @var float
+     */
+    private $itemsPerPageByDefault = 30.;
```

Comment:
For the number of items by page OK. But for the numbers of page it can happen. It's not a coherent API (and will require casting) to have some properties has int and others has float. I'll take a further look at that max numbers etc) when I'll have acces to a computer.